src/atomate2/openff/utils.py

```diff
@@ -0,0 +1,304 @@
+    name : str, optional
+        The name of the molecule. If not provided, defaults to the SMILES string.
+    charge_scaling : float, optional
+        The scaling factor for partial charges. Default is 1.
+    charge_method : str, optional
+        The charge method to use if partial charges are not provided. If not specified,
+        defaults to "custom" if partial charges are provided, else "am1bcc".
+    geometry : Union[pymatgen.core.Molecule, str, Path], optional
+        The geometry to use for adding conformers. Can be a Pymatgen Molecule, file path
+        or None.
+    partial_charges : List[float], optional
+        A list of partial charges to assign, or None to use the charge method.
+
+    Returns
+    -------
+    MoleculeSpec
+        The created MoleculeSpec
+    """
+    if charge_method is None:
+        charge_method = "custom" if partial_charges is not None else "am1bcc"
+
+    openff_mol = create_openff_mol(
+        smiles,
+        geometry,
+        charge_scaling,
+        partial_charges,
+        charge_method,
+    )
+
+    # create mol_spec
+    return MoleculeSpec(
+        name=(name or smiles),
+        count=count,
+        charge_scaling=charge_scaling,
+        charge_method=charge_method,
+        openff_mol=openff_mol.to_json(),
+    )
+
+
+def create_mol_spec_list(
+    input_mol_specs: list[MoleculeSpec | dict],
+) -> list[MoleculeSpec]:
+    """
+    Coerce and sort a MoleculeSpecs and dicts to MoleculeSpecs.
+
+    Will sort alphabetically based on concatenated smiles and name.
+
+    Parameters
+    ----------
+    input_mol_specs : list[dict | MoleculeSpec]
+        List of dicts or MoleculeSpecs to coerce and sort.
+
+    Returns
+    -------
+    List[MoleculeSpec]
+        List of MoleculeSpecs sorted by smiles and name.
+    """
+    mol_specs = []
+
+    for spec in input_mol_specs:
+        if isinstance(spec, dict):
+            mol_specs.append(create_mol_spec(**spec))
+        elif isinstance(spec, MoleculeSpec):
+            mol_specs.append(copy.deepcopy(spec))
+        else:
+            raise TypeError("mol_specs must be a list of dicts or MoleculeSpec")
+
+    mol_specs.sort(
+        key=lambda x: tk.Molecule.from_json(x.openff_mol).to_smiles() + x.name
+    )
+
+    return mol_specs
+
+
+def merge_specs_by_name_and_smiles(mol_specs: list[MoleculeSpec]) -> list[MoleculeSpec]:
+    """Merge MoleculeSpecs with the same name and SMILES string.
+
+    Groups MoleculeSpecs by their name and SMILES string, and merges the counts of specs
+    with matching name and SMILES. Returns a list of unique MoleculeSpecs.
+
+    Parameters
+    ----------
+    mol_specs : List[MoleculeSpec]
+        A list of MoleculeSpecs to merge.
+
+    Returns
+    -------
+    List[MoleculeSpec]
+        A list of merged MoleculeSpecs with unique name and SMILES combinations.
+    """
+    mol_specs = copy.deepcopy(mol_specs)
+    merged_spec_dict: dict[tuple[str, str], MoleculeSpec] = {}
```

Comment:
could use `merged_spec_dict=defaultdict(int)` to avoid the `if key in merged_spec_dict`

Reply:
My preference is to leave as is, the values are tuples with no sensible default